x-pack/plugins/transform/server/routes/api/transforms_stats_single/register_route.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  transformIdParamSchema,
+  type TransformIdParamSchema,
+} from '../../../../common/api_schemas/common';
+import { addInternalBasePath } from '../../../../common/constants';
+
+import type { RouteDependencies } from '../../../types';
+
+import { routeHandler } from './route_handler';
+
+export function registerRoute({ router, license }: RouteDependencies) {
+  /**
+   * @apiGroup Transforms
+   *
+   * @api {get} /internal/transform/transforms/:transformId/_stats Get transform stats
```

Comment:
As discussed, would be a good opportunity to add an API test for this route.

Reply:
Added the API integration tests in [273ff0f](https://github.com/elastic/kibana/pull/171201/commits/273ff0fcb4f5da449543a608529ef99de163b569). Had to refactor the whole file a bit because asserting 2 transforms was hardcoded. Updated the assertion helper function to work with different expectations.